web3c/signer.js

```diff
@@ -0,0 +1,42 @@
+const nacl = require('tweetnacl');
+const hash = require('js-sha512').sha512_256;
+const bytes = require('./bytes');
+
+/**
+ * Given public key, a signer can verify web3c signatures over public keys from the public key.
+ */
+class Signer {
+
+  constructor(publicKey) {
+	this.publicKey = publicKey;
+  }
+
+  /**
+   * Ensures the signature is Sign_{key_manager}(hash(longTermKey || timestamp)).
+   *
+   * @param  {String} signature is a hex string signature of keyHex || timestamp.
+   * @param  {String} keyHex is a hex string of the long term public key.
+   * @param  {Number?} timestamp is the timestamp at which the key expires.
+   * @throws {Error} if signature !== Sign_{KEY_MANAGER_PUBLIC_KEY}(longTermKey || timestamp).
```

Comment:
This interface wasn't super clear to understand reading through this up to here - can we return a an error or null, so that it's clear how `verify` is used when it's called - that the caller ha is responsible for turning problems into a thrown error if they believe the context is fatal?

Reply:
Agreed this makes for a better interface.

Reply:
Done.